MechJeb2/MechJebModuleGuidanceController.cs

```diff
@@ -242,7 +242,7 @@ public void flightangle4constraint(double rT, double vT, double inc, double gamm
                 Debug.Log("[MechJeb] MechJebModuleGuidanceController: setting up flightangle4constraint, rT: " + rT + " vT:" + vT + " inc:" + inc + " gamma:" + gamma);
                 PontryaginLaunch solver = NewPontryaginForLaunch(inc, sma);
                 solver.fixedCoast = fixedCoast;
-                solver.flightangle4constraint(rT, vT, gamma * UtilMath.Deg2Rad, inc * UtilMath.Deg2Rad);
+                solver.flightangle4constraint(rT, vT, gamma * UtilMath.Deg2Rad, Math.Abs(inc * UtilMath.Deg2Rad));
```

Comment:
Inclination, as an angle between two planes, is normally defined to always be positive and between 0° and 180°; what MJ does is really just sticking a boolean flag into the sign bit of the value to avoid having to pass it around separately. That is the reason why I removed this flag at the earliest opportunity, at a location where there already happened to be a unit conversion. This keeps the math for the Pontraygin methods as close to the original formulation as possible.

Reply:
Yeah, but what I'm thinking of is making it the responsibility of the constructor:

https://github.com/MuMech/MechJeb2/blob/dev/MechJeb2/Pontryagin/PontryaginLaunch.cs#L31=

So throw a Math.Abs() on that line, and all the similar lines, to normalize it and make that API correct with what you throw at it.  That doesn't change the math later (so I wouldn't want to start throwing Math.Abs() calls into line 44).  Constructors are an appropriate place to deal with building "permissive in what you accept" APIs.  Otherwise the constructor should really throw if the inc is negative since that isn't valid.

Reply:
(uh well its not really a constructor, but at some point in the future those probably need to be busted up into objects and that code needs to be a constructor, keep forgetting which version of my rewrites is actually live in this codebase and what is just stuff i've played around with or is in my head...)

Reply:
Done.